craft_application/commands/lifecycle.py

```diff
@@ -420,6 +422,55 @@ def _run(
         return self._run_real(parsed_args=parsed_args, step_name=step_name)
 
 
+class TestCommand(PackCommand):
+    """Command to run project tests.
+
+    The test command invokes the spread command with a processed spread.yaml
+    configuration file.
+
+    This command is opt-in for applications in craft-application 5 and will become
+    a standard lifecycle command in a future major release.
+    """
+
+    name = "test"
+    help_msg = "Run project tests"
+    overview = textwrap.dedent(
+        """
+        Run spread tests for the project.
+        """
+    )
+    common = True
+
+    @override
+    def _fill_parser(self, parser: argparse.ArgumentParser) -> None:
+        # Skip the parser additions that `pack` adds.
+        super(LifecycleCommand, self)._fill_parser(parser)
+
+    @override
+    def _run(
+        self,
+        parsed_args: argparse.Namespace,
+        step_name: str | None = None,
+        **kwargs: Any,
+    ) -> None:
+        # Output into the spread directory.
+        parsed_args.output = pathlib.Path.cwd() / "spread"
```

Comment:
q for the future: should we remove old/stale packages from this output dir?

Reply:
This is a hard to answer, as I don't know how we'd reliably do it unless we delete the file at the end of the test run. We could do that, but I'm not sure we want to.

Reply:
I think we have some decision about ergonomics to make in the near-ish future. For example, if I understand this code correctly then first all pack-able packages will be built first, and then test will run. Which means that the spread tests/setup will have to account for the fact that there are multiple artifacts to test